### ISSUE TITLE
[Enhancement] Bandwidth query parameters

### ISSUE THREAD

**Issue by garethsb** [AMWA-TV/nmos-network-control#24]
Bandwidths in the Network Control API are given as "\<number\>\<unit\>" strings in the JSON resource representations.

The RAML `queryParameters` seem currently not well defined.

What constitutes an exact match?
* Does "4Gbit/s" match "4.0Gbit/s"?
* How about "4000Mbit/s"?
* How about "4.000000001Gbit/s"?

(Aside: RQL didn't make the API v1.0, but 'eq', 'gt', 'lt', etc. would be useful with bandwidths.)

**Comment by vsachin33** [comment]
N/W controller is expected to set up  flows based on BW indicated in endpoint object and not under/over provision. What would be use case(s) for gt or lt?

 

Not sure I understand exact match question.  N/W controller needs to support BW based on corresponding schema and support all these formats ..Sachin

**Comment by garethsb** [comment]
Given the following data in the Network Controller `/endpoints`:

```json
[
  {
    "id": "fa08a879-488d-4e2b-8104-805e2288d8e9", ...
    "max_bandwidth": "10Gbit/s",
  },
  {
    "id": "d9c88a0a-c0c8-4ec8-ade5-fde9106e5040", ...
    "max_bandwidth": "10.0Gbit/s",
  },
  {
    "id": "e90af02c-03ff-4a43-9cb9-4d86e0f3161a", ...
    "max_bandwidth": "0010Gbit/s",
  }
]
```

What does the request `/endpoints?max_bandwidth=10Gbit/s` return?

**Comment by andrewbonney** [comment]
Based upon other APIs, basic queries would expect an exact match, so that specific query would only return one of those results. Personally I think this would be better resolved via the notes in #31 rather than changing the query language to handle complex translations.

**Comment by garethsb** [comment]
Agreed
